File: 1971/main.go
package main

func validPath(n int, edges [][]int, source int, destination int) bool {
	if n == 1 {
		return true
	}
	var stack []int
	for i := 0; i < len(edges); i++ {
		if edges[i][0] == source || edges[i][1] == source {
			if edges[i][0] == destination || edges[i][1] == destination {
				return true
			} else {
				if edges[i][0] == source {
					stack = append(stack, edges[i][1])
				} else {
					stack = append(stack, edges[i][0])
				}
			}
			edges = append(edges[:i], edges[i+1:]...)
			i--
		}
	}
	for _, s := range stack {
		if validPath(n, edges, s, destination) {
			return true
		}
	}
	return false
}

File: 1971/main.py
class Solution:
    def validPath(self, n: int, edges: list[list[int]], source: int, destination: int) -> bool:
        stack = set()
        i=0
        while i<len(edges):
            if source in edges[i]:
                for e in edges[i]:
                    if e == destination:
                        return True
                    if e != source:
                        stack.add(e)
                edges.pop(i)
            else:
                i+=1
        for sor in stack:
            if self.validPath(n, edges, sor, destination):
                return True
        return False        
            
        # edges_dic = {n: edges[n] for n in edges}
        # found = False
        # stack = [source]
        # while stack:
        #     for edge in edges[stack[0]]:
        #         if edge == destination:
        #             return True
        #         if edges_dic.get(stack[0], False):
    
    
                    
